tests/autoyast/installation.pm

```diff
@@ -173,15 +175,15 @@ sub run {
     $checktime = 30;
     $looptime  = 0;
     $timeout   = 0;
-    while (!$timeout) {
+    while ( !$timeout ) {
 
         mouse_hide(1);
-        $ret = check_screen(["reboot-after-installation", "autoyast-error"], $checktime);
+        $ret = check_screen( [ "reboot-after-installation", "autoyast-error" ], $checktime );
 
         #repeat until timeout or login screen
-        if (defined $ret) {
-            if (match_has_tag('autoyast-error')) {
-                record_soft_failure 'AUTOYAST_EXPECT_ERRORS ' . get_var('AUTOYAST_EXPECT_ERRORS_REASON');
+        if ( defined $ret ) {
+            if ( match_has_tag('autoyast-error') ) {
+                diag( 'Expected error. Iteration = ' . $i );
```

Comment:
1. You basically trying to move out soft failure from git , so adding / removing soft failure from the test can be done without doing PR and discussing this with at least one more person which is bad
2. Such test will be hardlinked to some certain environment  if someone will try to create test suite for this scenario in his personal machine he will suppose to be aware about variable AUTOYAST_EXPECT_ERRORS_REASON and will need to copy it current value from osd / o3 otherwise he/she will get not same result and this is bad 
3. Actually I fail to understand expected flow when I will have more than one error reason how code suppose to handle this ?